app/views/spaces_new.erb

```diff
@@ -1,31 +1,31 @@
 <% @title="List a Space | MakersBnB" %>
 
-<h2>List a Space</h2>
+<h2><%= @space ? "Update Space" : "List a Space" %></h2>
 <form action="/spaces" method="post">
   <label for="name">
     Name:
-    <input type="text" name="name" required>
+    <input type="text" name="name" value="<%= @space.name if @space %>" required>
   </label>
 
   <label for="description">
     Description:
-    <textarea name="description" cols="30" rows="10"></textarea>
+    <textarea name="description" cols="30" rows="10"><%= @space.description if @space %></textarea>
   </label>
 
   <label for='price'>
      Price:
-    <input type='number' name='price' step="0.01" required/>
+    <input type='number' name='price' step="0.01" value="<%= @space.price if @space %>" required/>
   </label>
 
   <label for='date_from'>
       Available From:
-      <input type='date' name='date_from' required/>
+      <input type='date' name='date_from' value="<%= @space.date_from if @space %>" required/>
   </label>
 
   <label for="date_to">
       Available To:
-      <input type='date' name='date_to' required/>
+      <input type='date' name='date_to' value="<%= @space.date_to if @space %>" required/>
   </label>
 
-  <input type="submit" value="List a Space">
+  <input type="submit" value="<%= @space ? "Update Space" : "List a Space" %>">
 </form>
```

Comment:
just trying to understand this, is this pre-populating the form fields with the existing space details if a user is going to amend the details?